src/clients/scc.rb

```diff
@@ -51,6 +56,16 @@ def main
           ::Registration::SwMgmt.init
 
           return WFM.call("inst_scc", WFM.Args)
+        rescue Registration::SourceRestoreError => e
+          # TRANSLATORS: Error message in RichText format, %s contains the details from libzypp
+          Report.LongError(_("<p>The repository initialization failed. " \
+            "Disable (or remove) the offending service or repository " \
+            "in the repository manager.</p><p>Details:</p><p>%s</p>") % CGI.escapeHTML(e.message))
+          ret = WFM.call("repositories", WFM.Args)
+          log.info "repository manager result: #{ret}"
+          # drop all loaded repos, force complete reloading
+          Pkg.SourceFinishAll
+          retry if ret == :next
```

Comment:
well, at first, only SwMgmt.init should be enclosed and the second note, is that now method start to be quite long. So it should be shortened like:

```
def print_help
  cmdline_description = {
     "id"   => "scc",
     # Command line help text for the repository module, %1 is "SUSEconnect"
     "help" => _("Use '%s' instead of this YaST module.") % "SUSEconnect"
   }
 
   CommandLine.Run(cmdline_description)
end

def fix_repos
  # TRANSLATORS: Error message in RichText format, %s contains the details from libzypp
   Report.LongError(_("<p>The repository initialization failed. " \
     "Disable (or remove) the offending service or repository " \
     "in the repository manager.</p><p>Details:</p><p>%s</p>") % CGI.escapeHTML(e.message))
    ret = WFM.call("repositories", WFM.Args)
    log.info "repository manager result: #{ret}"
    # drop all loaded repos, force complete reloading
     Pkg.SourceFinishAll
     ret == :next
end

def main
       textdomain "registration"
 
       if WFM.Args.include?("help")
         print_help
       else
         Wizard.CreateDialog
 
         begin
           ::Registration::SwMgmt.init
 
           return WFM.call("inst_scc", WFM.Args)
         rescue Registration::SourceRestoreError => e
           retry if fix_repos
         ensure
           Wizard.CloseDialog
         end
end
```

btw that reminds me, that we should have safe block variant for Wizard.CreateDialog.